source/extensions/filters/http/ratelimit/ratelimit_headers.cc

```diff
@@ -0,0 +1,82 @@
+#include "extensions/filters/http/ratelimit/ratelimit_headers.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "absl/strings/substitute.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitFilter {
+
+Http::ResponseHeaderMapPtr XRateLimitHeaderUtils::create(
+    Filters::Common::RateLimit::DescriptorStatusListPtr&& descriptor_statuses) {
+  Http::ResponseHeaderMapPtr result = Http::ResponseHeaderMapImpl::create();
+  if (!descriptor_statuses || descriptor_statuses->empty()) {
+    descriptor_statuses = nullptr;
+    return result;
+  }
+
+  absl::optional<envoy::service::ratelimit::v3::RateLimitResponse_DescriptorStatus>
+      min_remaining_limit_status;
+  std::string quota_policy;
+  for (auto&& status : *descriptor_statuses) {
+    if (!status.has_current_limit()) {
+      continue;
+    }
+    if (!min_remaining_limit_status ||
+        status.limit_remaining() < min_remaining_limit_status.value().limit_remaining()) {
+      min_remaining_limit_status.emplace(status);
+    }
+    const uint32_t window = convertRateLimitUnit(status.current_limit().unit());
+    // Constructing the quota-policy per RFC
+    // https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html#rfc.section.3.1
```

Comment:
Reference latest draft.

```suggestion
    // https://tools.ietf.org/id/draft-polli-ratelimit-headers-02.html#name-quota-policy
```

I suggest implementing the mandatory part of the spec first (without the quota policy) and add the quota policy information on a subsequent PR.